subsys/modbus/modbus_server.c

```diff
@@ -111,11 +111,11 @@ static bool mbs_fc01_coil_read(struct modbus_context *ctx)
 	uint8_t *presp;
 	bool coil_state;
 	int err;
-	uint16_t coil_addr;
-	uint16_t coil_qty;
-	uint16_t num_bytes;
-	uint8_t bit_mask;
-	uint16_t coil_cntr;
+	uint_fast16_t coil_addr;
+	uint_fast16_t coil_qty;
+	uint_fast16_t num_bytes;
+	uint_fast8_t bit_mask;
+	uint_fast16_t coil_cntr;
```

Comment:
Not only do I see no reason to optimize this, which always uses one more resource, but it is not guaranteed by standard to be the fastest. Here we have protocol handling, and I prefer exact integer types for that not some integer.

Reply:
Acknowledging the importance of precision in protocol handling, I evaluated the impact of using `uint_fast8_t` and `uint_fast16_t` within `samples\subsys\modbus\rtu_server`, finding it could reduce the code footprint by around 100 bytes on both `frdm_k64f (Cortex-M4)` and `gd32vf103c_starter (RISC-V)` platforms.
These types align with the processor's natural word size, potentially enhancing performance and efficiency.

If the benefits of this optimization don't justify the change, and there's a preference for strictly using `uint8_t` and `uint16_t` for local variables, I'm willing to revoke this commit: https://github.com/zephyrproject-rtos/zephyr/pull/68505/commits/1859018dd71c89fa739173502128789f04c8211e.